app/helpers/hearing_paginator.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+# Creates a chronologically sorted set of "items"
+# with next and previous navigation helpers.
+#
+# Next and previous correlate to navigation over all
+# a cases' hearing's hearing days.
+#
+class HearingPaginator
+  include ActionView::Helpers
+  include Rails.application.routes.url_helpers
+
+  PageItem = Struct.new(:id, :hearing_date)
+
+  # rubocop:disable Rails/HelperInstanceVariable
```

Comment:
somewhat bizarrely memoization is explicitly excluded by this cop as a result, apparently of this accepted issue: https://github.com/rubocop-hq/rubocop-rails/issues/311. I was thinking to raise my own issue related to using a class instance within a helper.

Reply:
...and in case you are thinking of the cheeky ;) workaround below
```
@current_page ||= page.to_i
```

it raises another cop violation
```
Lint/DisjunctiveAssignmentInConstructor: Unnecessary disjunctive assignment. Use plain assignment.
```

Reply:
You can have `@current_page = page.to_i` in the initializer as it it known to be `nil` at that point. In the `current_page` method below (now lines 20-22) the check (I think) is not necessary as it has already been set.